libraries/dagster-delta/dagster_delta/_handler/base.py

```diff
@@ -53,6 +53,50 @@ def get_output_stats(self, obj: T) -> dict[str, MetadataValue]:
         """Abstract method to return output stats"""
         pass
 
+    def _find_keys_in_metadata(
+        self,
+        context: OutputContext,
+        keys=["merge_predicate", "merge_operations_config"],
+    ) -> Dict[str, Any]:
+        """Finds the keys in the metadata in the following order:
+
+        It will find the merge_predicate or merge_operations_config in this order:
+        1. Runtime metadata
+        2. Definition metadata
+        3. IO Manager config
+
+        E.g., `merge_predicate` and `merge_operations_config`
+
+        Args:
+            context (OutputContext): The output context
+            keys (list[str], optional): The keys to find in the metadata. Defaults to ["merge_predicate", "merge_operations_config"].
+
+        Returns:
+            dict[str, Any]: The metadata with the keys found
+        """
+        metadata_definition = context.definition_metadata or {}
+        metadata_output = context.output_metadata or {}
+
+        # Find each of the key in the definition or output metadata
+        result = {}
+
+        for key in keys:
+            if key in metadata_output or {}:
+                result[key] = metadata_output[key]
+            elif key in metadata_definition or {}:
+                result[key] = metadata_definition[key]
+            else:
+                result[key] = None
+
+            # If it's a TextMetadataValue, cast it to string
+            if isinstance(result[key], MetadataValue):
+                if result[key].value is not None:
+                    result[key] = str(result[key].value)
+                else:
+                    result[key] = None
```

Comment:
```suggestion
            if isinstance(result[key], MetadataValue):
                if result[key].value is not None:
                    result[key] = result[key].value
```
This should be enough